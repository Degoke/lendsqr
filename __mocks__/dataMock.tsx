import UsersIcon from '../public/images/users_card_icon.svg'

export const mockCard = {
  image: UsersIcon,
  text: 'Users',
  amount: 100,
}

export const mockImage = UsersIcon

export const MockComponent = () => (
  <div data-testid="mockComponent">Hello World</div>
)

export const mockList = {
  title: 'settings',
  items: [
    {
      text: 'preferences',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'fees and pricing',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'audit logs',
      url: '',
      icon: UsersIcon,
    },
  ],
}
