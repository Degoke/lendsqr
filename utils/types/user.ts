export interface IUser {
  id: string
  createdAt: string
  orgName: string
  userName: string
  email: string
  phoneNumber: string
  lastActiveDate: string
  profile: UserProfile
  guarantor: UserGuarantor
  accountBalance: string
  accountNumber: string
  socials: UserSocials
  education: UserEducation
  status?: UserStatus
}

export type UserStatus = 'inactive' | 'pending' | 'blacklisted' | 'active'

interface UserProfile {
  firstName: string
  lastName: string
  phoneNumber: string
  avatar: string
  gender: string
  bvn: string
  address: string
  currency: string
}

interface UserGuarantor {
  firstName: string
  lastName: string
  phoneNumber: string
  gender: string
  address: string
}

interface UserSocials {
  facebook: string
  instagram: string
  twitter: string
}

interface UserEducation {
  level: string
  employmentStatus: string
  sector: string
  duration: string
  officeEmail: string
  monthlyIncome: string[]
  loanRepayment: string
}

export type UserFilters =
  | 'orgName'
  | 'email'
  | 'userName'
  | 'createdAt'
  | 'status'
  | 'phoneNumber'
