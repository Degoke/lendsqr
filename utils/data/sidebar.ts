import { urls } from '../constants/urls'
import { IListItem } from '../types/list'
import Users from '../../public/images/users.svg'
import Guarantors from '../../public/images/guarantors.svg'
import Loans from '../../public/images/loans.svg'
import Decision from '../../public/images/decision.svg'
import Savings from '../../public/images/savings.svg'
import Whitelist from '../../public/images/whitelist.svg'
import Karma from '../../public/images/karma.svg'
import Products from '../../public/images/products.svg'
import Fees from '../../public/images/fees.svg'
import Transactions from '../../public/images/transactions.svg'
import Services from '../../public/images/services.svg'
import Service from '../../public/images/service.svg'
import Settlements from '../../public/images/settlements.svg'
import Reports from '../../public/images/reports.svg'
import Pricing from '../../public/images/pricing.svg'
import Audit from '../../public/images/audit.svg'
import Requests from '../../public/images/requests.svg'
import Organization from '../../public/images/organization.svg'
import Preferences from '../../public/images/preferences.svg'

interface ISideBarData {
  title: string
  items: IListItem[]
}

export const sidebarData: ISideBarData[] = [
  {
    title: 'customers',
    items: [
      {
        text: 'users',
        url: urls.USERS,
        icon: Users,
      },
      {
        text: 'guarantors',
        url: '',
        icon: Guarantors,
      },
      {
        text: 'loans',
        url: '',
        icon: Loans,
      },
      {
        text: 'decision models',
        url: '',
        icon: Decision,
      },
      {
        text: 'savings',
        url: '',
        icon: Savings,
      },
      {
        text: 'loan requests',
        url: '',
        icon: Requests,
      },
      {
        text: 'whitelist',
        url: '',
        icon: Whitelist,
      },
      {
        text: 'karma',
        url: '',
        icon: Karma,
      },
    ],
  },
  {
    title: 'businesses',
    items: [
      {
        text: 'organization',
        url: '',
        icon: Organization,
      },
      {
        text: 'loan products',
        url: '',
        icon: Loans,
      },
      {
        text: 'savings products',
        url: '',
        icon: Products,
      },
      {
        text: 'fees and charges',
        url: '',
        icon: Fees,
      },
      {
        text: 'transactions',
        url: '',
        icon: Transactions,
      },
      {
        text: 'services',
        url: '',
        icon: Services,
      },
      {
        text: 'service accounts',
        url: '',
        icon: Service,
      },
      {
        text: 'settlements',
        url: '',
        icon: Settlements,
      },
      {
        text: 'reports',
        url: '',
        icon: Reports,
      },
    ],
  },
  {
    title: 'settings',
    items: [
      {
        text: 'preferences',
        url: '',
        icon: Preferences,
      },
      {
        text: 'fees and pricing',
        url: '',
        icon: Pricing,
      },
      {
        text: 'audit logs',
        url: '',
        icon: Audit,
      },
    ],
  },
]
