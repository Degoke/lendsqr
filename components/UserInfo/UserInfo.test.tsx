import { render, screen } from '@testing-library/react'
import PersonalInfo from './personalInfo'
import { mockCard, mockImage } from '../../__mocks__/dataMock'
import { mockUser } from '../../utils/data/user'
import UserEducation from './userEducation'
import UserGuarantor from './userGuarantor'
import UserSocials from './userSocials'

describe('PersonalInfo tests', () => {
  it('renders PersonalInfo component unchanged', () => {
    const { container } = render(<PersonalInfo user={mockUser[0]} />)
    expect(container).toMatchSnapshot()
  })
})

describe('UserEducation tests', () => {
  it('renders UserEducation component unchanged', () => {
    const { container } = render(<UserEducation user={mockUser[0]} />)
    expect(container).toMatchSnapshot()
  })
})

describe('UserGuarantor tests', () => {
  it('renders UserGuarantor component unchanged', () => {
    const { container } = render(<UserGuarantor user={mockUser[0]} />)
    expect(container).toMatchSnapshot()
  })
})

describe('UserSocials tests', () => {
  it('renders UserSocials component unchanged', () => {
    const { container } = render(<UserSocials user={mockUser[0]} />)
    expect(container).toMatchSnapshot()
  })
})
