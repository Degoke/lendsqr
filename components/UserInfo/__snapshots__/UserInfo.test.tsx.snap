// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalInfo tests renders PersonalInfo component unchanged 1`] = `
<div>
  <div
    class="userInfo"
  >
    <h2>
      Personal Information
    </h2>
    <div
      class="userInfo__information"
    >
      <div
        class="userInfo__infoGroup"
      >
        <p>
          Full name
        </p>
        <p>
          Darian Rolfson
        </p>
      </div>
      <div
        class="userInfo__infoGroup"
      >
        <p>
          phone number
        </p>
        <p>
          (553) 208-0727 x31321
        </p>
      </div>
      <div
        class="userInfo__infoGroup"
      >
        <p>
          email address
        </p>
        <p>
          Maverick.Hyatt83@gmail.com
        </p>
      </div>
      <div
        class="userInfo__infoGroup"
      >
        <p>
          bvn
        </p>
        <p>
          815809412
        </p>
      </div>
      <div
        class="userInfo__infoGroup"
      >
        <p>
          gender
        </p>
        <p>
          Male
        </p>
      </div>
      <div
        class="userInfo__infoGroup"
      >
        <p>
          marital status
        </p>
        <p>
          -
        </p>
      </div>
      <div
        class="userInfo__infoGroup"
      >
        <p>
          children
        </p>
        <p>
          None
        </p>
      </div>
      <div
        class="userInfo__infoGroup"
      >
        <p>
          Type of Residence
        </p>
        <p>
          Gusikowski Locks
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`UserEducation tests renders UserEducation component unchanged 1`] = `
<div>
  <div
    class="userInfo"
  >
    <h2>
      Education and Employment
    </h2>
    <div
      class="userInfo__information"
    >
      <div
        class="userInfo__infoGroup"
      >
        <p>
          Level of education
        </p>
        <p>
          Bsc
        </p>
      </div>
      <div
        class="userInfo__infoGroup"
      >
        <p>
          employment status
        </p>
        <p>
          Employed
        </p>
      </div>
      <div
        class="userInfo__infoGroup"
      >
        <p>
          sector of employment
        </p>
        <p>
          FinTech
        </p>
      </div>
      <div
        class="userInfo__infoGroup"
      >
        <p>
          duration of employment
        </p>
        <p>
          2 Years
        </p>
      </div>
      <div
        class="userInfo__infoGroup"
      >
        <p>
          office email
        </p>
        <p>
          Edna4@yahoo.com
        </p>
      </div>
      <div
        class="userInfo__infoGroup"
      >
        <p>
          monthly income
        </p>
        <p>
          ₦118.07 - ₦128.57
        </p>
      </div>
      <div
        class="userInfo__infoGroup"
      >
        <p>
          loan repayment
        </p>
        <p>
          122.47
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`UserGuarantor tests renders UserGuarantor component unchanged 1`] = `
<div>
  <div
    class="userInfo"
  >
    <h2>
      Guarantor
    </h2>
    <div
      class="userInfo__information"
    >
      <div
        class="userInfo__infoGroup"
      >
        <p>
          Full name
        </p>
        <p>
          Celine Monahan
        </p>
      </div>
      <div
        class="userInfo__infoGroup"
      >
        <p>
          phone number
        </p>
        <p>
          1-482-227-3654 x71086
        </p>
      </div>
      <div
        class="userInfo__infoGroup"
      >
        <p>
          email address
        </p>
        <p>
          O'Hara Centers
        </p>
      </div>
      <div
        class="userInfo__infoGroup"
      >
        <p>
          relationship
        </p>
        <p>
          Male
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`UserSocials tests renders UserSocials component unchanged 1`] = `
<div>
  <div
    class="userInfo"
  >
    <h2>
      Socials
    </h2>
    <div
      class="userInfo__information"
    >
      <div
        class="userInfo__infoGroup"
      >
        <p>
          Twitter
        </p>
        <p>
          @lendsqr
        </p>
      </div>
      <div
        class="userInfo__infoGroup"
      >
        <p>
          Facebook
        </p>
        <p>
          @lendsqr
        </p>
      </div>
      <div
        class="userInfo__infoGroup"
      >
        <p>
          Instagram
        </p>
        <p>
          @lendsqr
        </p>
      </div>
    </div>
  </div>
</div>
`;
